pkg/cluster/k8sres.go

```diff
@@ -961,13 +961,15 @@ func (c *Cluster) generateService(role PostgresRole, spec *acidv1.PostgresSpec)
 		// safe default value: lock load balancer to only local address unless overridden explicitly.
 		sourceRanges := []string{localHost}
 
-		allowedSourceRanges := spec.AllowedSourceRanges
-		if len(allowedSourceRanges) >= 0 {
-			sourceRanges = allowedSourceRanges
+		// spec.AllowedSourceRanges evaluates to the empty slice of zero length
+		// when omitted or set to 'null'/empty sequence in the PG manifest
+		if len(spec.AllowedSourceRanges) > 0 {
+			sourceRanges = spec.AllowedSourceRanges
```

Comment:
Minor nitpick, but it seems we don't need the sourceRanges variable altogether, i.e
```
if len(spec.AllowedSourceRanges) > 0 {
   serviceSpec.LoadBalancerSourceRanges = spec.AllowedSourceRanges
} else {
   serviceSpec.LoadBalancerSourceRanges = []string{localHost}
}
```